templates/host/networks/network.xml

```diff
@@ -10,6 +10,11 @@
     {%   for h in hosts|sort(attribute="ip") if h.fqdn is defined or h.host is defined -%}
     <host ip='{{ h.ip }}'><hostname>{{ h.fqdn if h.fqdn is defined else '%s.%s' % (h.host, domain) }}</hostname></host>
 {%       endfor %}
+{%- if dns is defined %}
+    {%   for d in dns -%}
```

Comment:
I think that we have to make sure that d.{ip,hostname} are defined and not empty such like:
  s/for d in dns/& if d.ip and d.hostname/

Reply:
Except for that LGTM. Once I'll merge your PR and will update a little ASAP.